velox/functions/prestosql/JsonFunctions.cpp

```diff
@@ -63,10 +62,61 @@ class JsonFormatFunction : public exec::VectorFunction {
   }
 };
 
+class JsonParseFunction : public exec::VectorFunction {
+ public:
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    VectorPtr localResult;
+
+    // Input can be constant or flat.
+    assert(args.size() > 0);
+    const auto& arg = args[0];
+    if (arg->isConstantEncoding()) {
+      auto value = arg->as<ConstantVector<StringView>>()->valueAt(0);
+      folly::parseJson(value);
```

Comment:
This function has a high per-row cost and there are no particular optimizations that would make vector function more efficient than simple function. Hence, I suggest to use simple function API to implement this function.

folly::parseJson is particularly expensive because it creates a folly::dynamic object that represents the JSON string and in doing so allocated lots of objects. Since we do not use the resulting object(s), we should implement parsing logic that doesn't do any memory allocation. All we need is to validate that JSON string is valid. If folly doesn't provide a lightweight validation function, let's write our own.

CC: @laithsakka 

P.S. Not directly related to this PR, but FYI: https://github.com/facebookincubator/velox/issues/3658

Reply:
For now we have included a todo on the function that will be taken up during performance improvements. So we will use this implementation to get unblocked.